src/AutoRest.CSharp/Common/Input/InputTypes/InputDurationType.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace AutoRest.CSharp.Common.Input;
+
+internal record InputDurationType(DurationKnownEncoding Encode, InputPrimitiveType WireType, bool IsNullable) : InputType("Duration", IsNullable)
+{
+    public static InputDurationType Iso8601 { get; } = new(DurationKnownEncoding.Iso8601, InputPrimitiveType.String, false);
```

Comment:
why are these always not nullable?

Reply:
these are helpers for convenience usage. We do not have to have these.
I am just following the pattern we have in `InputPrimitiveType`.
If you want a nullable version, just call the ctor which is public as well

Reply:
I think it's OK to have the helpers, is it possible to turn them into methods to input more parameters such as nullable.
I am a bit concerned that with this in place, the consumer would simply ignore the nullability for them.

Reply:
To avoid further confusions, I will remove these.